### ISSUE TITLE
Error: access violation accessing 0x0 when calling Java.enumerateLoadedClasses() or Java.choose()

### ISSUE THREAD

**Issue by StefaanSeys** [frida/frida-java#42]
Hello,

When executing the following simple scripts
```
Java.perform(function () {   
        Java.enumerateLoadedClasses({
            onMatch: function(className){
                console.log(className)
            },
            onComplete:function(){
            }
    });
});
````
I get this error:
```
{'type': 'error', 
'description': 'Error: access violation accessing 0x0', 
'stack': 'Error: access violation accessing 0x0
    at v (frida/node_modules/frida-java/index.js:233)
    at frida/node_modules/frida-java/index.js:105
    at [anon] (script1.js:10)
    at frida/node_modules/frida-java/index.js:266
    at call (native)
    at dispatchMessage (input:1)', 'fileName': 'frida/node_modules/frida-java/index.js', 'lineNumber': 233, 'columnNumber': 1}
```

I get the same issue when calling **`Java.choose()`**.

Versions used:

- Android 4.3.1 (Cyanogenmod 10.2.1)
- Both with Frida version 9.1.28 and 10.5.13

**Comment by StefaanSeys** [comment]
I have move my test platform up to Android 4.4.4 (Cyanogenmod 11) and now the exact same script does work with both Frida versions.
This issue can be closed.

**Comment by Myonium** [comment]
There seems to be an issue with Android 4.3
Test set:
Samsung 3 
Android 4.3
Frida-Server: frida-server-10.6.2-android-arm
Frida-Client: frida 10.6.3 

On executing:
`Java.perform(function () {   
        Java.enumerateLoadedClasses({
            onMatch: function(className){
                console.log(className)
            },
            onComplete:function(){
            }
    });
});`
I receive the following error:
`Error: access violation accessing 0x0
    at v (frida/node_modules/frida-java/index.js:232)
    at frida/node_modules/frida-java/index.js:105
    at [anon] (input:8)
    at frida/node_modules/frida-java/lib/vm.js:39
    at frida/node_modules/frida-java/index.js:281
    at eval (input:9)
    at eval (native)
    at repl1.js:4
    at apply (native)
    [...]`

**Comment by audebert** [comment]
Same issue on:
- Android 4.3
- Samsung Galaxy Nexus
- Frida-server: frida-server-10.6.18-android-arm
- Frida-client: 10.6.18

Code:
```python
jscode = """
setTimeout(function(){
    Java.enumerateLoadedClasses({
        onMatch: function(className) {
            send(className);
        },
        onComplete: function() { }
    });
}, 0);
"""

script = process.create_script(jscode)
script.on('message', on_message)
script.load()
```

Error message:
`{'lineNumber': 229, 'fileName': 'frida/node_modules/frida-java/index.js', 'description': 'Error: access violation accessing 0x0', 'stack': 'Error: access violation accessing 0x0\n    at v (frida/node_modules/frida-java/index.js:229)\n    at frida/node_modules/frida-java/index.js:108\n    at [anon] (script1.js:9)\n    at frida/runtime/core.js:57', 'type': 'error', 'columnNumber': 1}`

**Comment by audebert** [comment]
Actually this is a duplicate of #7

**Comment by enovella** [comment]
Getting something very similar with latest Frida & r2frida (all in latest versions v12.2.12)

```sh
Unhandled message: {"type":"error","description":"Error: access violation accessing 0x0","stack":"Error: access violation accessing 0x0\n    at input:1\n    at frida/node_modules/frida-java/index.js:339\n    at input:1","fileName":"input","lineNumber":1,"columnNumber":1}
```

Any regression or something?

**Comment by enovella** [comment]
Same here with latest Frida and r2frida :(

```java
Unhandled message: {"type":"error","description":"Error: access violation accessing 0x0","stack":"Error: access violation accessing 0x0\n    at input:1\n    at frida/node_modules/frida-java/index.js:339\n    at input:1","fileName":"input","lineNumber":1,"columnNumber":1}
```

**Comment by enovella** [comment]
If I clear the cache and data of the app before starting the app, the error doesn't show up anymore. But this need to be done everytime before running Frida/r2frida.
